### PR TITLE
[SPARK-10122][PySpark][Streaming] Fix getOffsetRanges bug in PySpark-Streaming transform function

### PR DESCRIPTION
**Pull request by jerryshao**
Details of the bug and explanations can be seen in [SPARK-10122](https://issues.apache.org/jira/browse/SPARK-10122).

@tdas , please help to review.

### CODE REVIEW

**Comment by tdas** [comment]
LGTM. But @zsxwing could you take another look at this?

File: python/pyspark/streaming/dstream.py [hunk 1]
```diff
@@ -610,7 +610,10 @@ def __init__(self, prev, func):
         self.is_checkpointed = False
         self._jdstream_val = None
 
-        if (isinstance(prev, TransformedDStream) and
+        # Using type() to avoid folding the functions and compacting the DStreams which is not
+        # not strictly a object of TransformedDStream.
```
Comment (zsxwing) [inline python/pyspark/streaming/dstream.py]: nit: two `not`s

**Comment by zsxwing** [comment]
LGTM except one minor doc typo.

**Comment by tdas** [comment]
I will merge this to master and branch 1.5 (will take care of the `not` while committing).

**Pull request closed without merging**